modules/tinymce/src/plugins/autosave/main/ts/core/Storage.ts

```diff
@@ -13,13 +13,18 @@ import * as Events from '../api/Events';
 import * as Settings from '../api/Settings';
 
 const isEmpty = (editor: Editor, html?: string) => {
-  const forcedRootBlockName = editor.settings.forced_root_block;
-
-  html = Tools.trim(typeof html === 'undefined' ? editor.getBody().innerHTML : html);
-
-  return html === '' || new RegExp(
-    '^<' + forcedRootBlockName + '[^>]*>((\u00a0|&nbsp;|[ \t]|<br[^>]*>)+?|)<\/' + forcedRootBlockName + '>|<br>$', 'i'
-  ).test(html);
+  if(typeof html === 'undefined') {
+    return editor.dom.isEmpty(editor.getBody());
+  } else {
+    html = Tools.trim(html);
+
+    if(html === '') {
+      return true;
+    } else {
+      const fragment = new DOMParser().parseFromString(html, 'text/html');
```

Comment:
Missing import here, as you need `import { DOMParser } from '@ephox/dom-globals';` to use this.